packages/back-end/src/models/SdkConnectionModel.ts

```diff
@@ -69,13 +68,13 @@ const SDKConnectionModel = mongoose.model<SDKConnectionInterface>(
 );
 
 function addEnvProxySettings(proxy: ProxyConnection): ProxyConnection {
-  if (IS_CLOUD) return proxy;
-
   return {
     ...proxy,
-    enabled: PROXY_ENABLED,
-    host: PROXY_HOST_INTERNAL || PROXY_HOST_PUBLIC,
-    hostExternal: PROXY_HOST_PUBLIC || PROXY_HOST_INTERNAL,
+    enabled: proxy.enabled !== undefined ? proxy.enabled : PROXY_ENABLED,
+    host: proxy.host ? proxy.host : PROXY_HOST_INTERNAL || PROXY_HOST_PUBLIC,
+    hostExternal: proxy.hostExternal
+      ? proxy.hostExternal
+      : PROXY_HOST_PUBLIC || PROXY_HOST_INTERNAL,
```

Comment:
What's the purpose of this change?  I think this might break self-hosted users who are using a proxy with environment variables.

Reply:
I believe the original function breaks our "local proxy for cloud users" functionality: They need to be able to set `enabled` and `host` from the SDK Connection UI, but then this method overwrites these fields with env var defaults. And even if they aren't actually set in our cloud environment, we'd still overwrite with `""`. We need a way to only set these fields if nothing is being provided by `...proxy`.

I don't fully understand the point of `hostExternal` so I'm not confident that I got this part correct.